media/moviResources.md

```diff
@@ -11,63 +11,78 @@ Such a resource contains the following components:
+    0010  fix         totalTime        Length of media (in seconds)
+    0014  fix         frameRate        Frame rate (informational only)
+    0018  int16       frameWidht       Width of video
+    001A  int16       frameHeight      Height of video
+    001C  int16       gfxNumBits       For videos may be 0x0008, 0x000F or 0x0018, 0x0000 otherwise
+    001E  int16       isPalette        Palette presence in file. For videos 0x0001, 0x0000 otherwise
+    0020  int16       audioNumChans    Numbers of audiochannels 0x0000 - no audio, 0x0001 - mono, 0x0002 stereo
+    0022  int16       audioSampleSize  Audio sample size, 0x0001 - 8-bit, 0x0002 - 16-bit
+    0026  fix         audioSampleRate  Sample rate of audio, in Hz (0x2B770000 - 11127 Hz, 0x56EE0000 - 22254 Hz)
+    0030  uint8[216]  reserved         Reserved
+    0108  uint8[768]  palette          Palette
 
-    0000  [4]byte  Container tag: "MOVI"
-    0004  int32    Number of index entries
-    0008  int32    Size of index table
-    000C  int32    Size of content (all data entries)
-    0010  [3]byte  Length of media (in unit of 1/0x10000 seconds)
-    0013  byte     Unused (see below). Always 0x00
+If the index table size is greater than the total of index entry number times one entry size, the remaining bytes are zero.
 
-    0014  int32    Unknown.
+> The index table sizes come in multiples of 0x0400. The rough rule for recreating the existing index sizes is:
+> Required size less than 0x0400 -> Use 0x0400. Otherwise: Round up to next 0x0400 border and add another 0x0400.
 
-    0018  int16    Width of video
-    001A  int16    Height of video
+**Palette** (256 * 3 bytes)
 
-    001C  int16    Unknown. For videos 0x0008, 0x0000 otherwise
-    001E  int16    Unknown. For videos 0x0001, 0x0000 otherwise
-    0020  int16    Unknown. Always 0x0001
-    0022  int32    Unknown. Always 0x00000001
+The palette is always present, even if the MOVI contains only audio. In such cases all the colour channels are 0x00.
 
-    0026  int16    Sample rate of audio
+**Movie chunk Entry** (`MovieChunk` struct) (8 bytes)
 
-    0028  []byte   Unknown. Always 0x00 until full size of header
+    0000  uint32              Chunk info (bitfield)
+    0004  uint32   offset     Offset to data (from beginning)
 
-> Although the engine might treat the media length as 4 bytes, the index table entries provide only 3 for timestamps.
-> The total length of the media can't be longer than the last possible entry, so the header will have only 3 as well.
+First 4 bytes are uint32 bit field which follow structure:
 
-If the index table size is greater than the total of index entry number times one entry size, the remaining bytes are zero.
+    bits  00-23  24  time       Fixed point timestamp from movie start (in seconds)
+    bits  24-26   3  played     Has this chunk been clocked out?
+    bits  27-31   4  flags      Chunk type specific flags
+    bits  32-32   1  chunkType  Chunk type (see below)
```

Comment:
This is confusing:
* played should be marked to be only used internally, most likely this wouldn't be stored in the file
* flags is listed to have bits 27-31 (which is 5 bits), but listed as 4
* chunkType has only one bit, yet the list below goes from chunk type 0 to 5, requiring at least 3 bits.

Please rewrite and/or specify how to interpret this. While this is now more detailed to the source, it is also more confusing to reimplement independently. Remember, the intention is to be able to implement other tools that can work with the resources, not document how complex the original engine's implementation was :)